src/arrayMethodJoin.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (separator !== '' && separator !== undefined) {
```

Comment:
it would be better to set default value for the argument instead of doing this check for each iteration